### DISCUSSION TITLE
Can I use table, column name as non english? (with prisma db pull)

### DISCUSSION THREAD

**Discussion by sdg9670** [prisma/prisma#19253] [Q&A] [answered]
## Problem

Can I use table, column name as non english?
I tried to generate prisma schema with `prisma db pull`
But I got a lot of warning like this.

```
/// This model or at least one of its fields has comments in the database, and requires an additional setup for migrations: Read more: https://pris.ly/d/database-comments
model  {
  id Int @id @default(autoincrement())
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 한글 String? @map("한글") @db.Text
}
```

You can see that the above content is missing the model name and the column is commented.

## Suggested solution

I need to use non english generator.. so, are there an option to disable valid name?

**Comment by aqrln** [comment]
Yes, you can. That's why Prisma added the `@map` attribute for you here so you can provide a supported name for the Prisma Client while still keeping the original name in the database.

You need to uncomment this field, provide a new name that will be used in the schema file and in your code, like this:
```prisma
model Entity {
  id Int @id @default(autoincrement())
  latinName String? @map("한글") @db.Text
}
```
and regenerate your client with `prisma generate`.

**Reply by sdg9670** [reply]
Our organization used the Latin variable name in js code.
So, We don't want to generate the comment for latin by force.
Because, It bothers our to check the generated schema every time.
I think It would be batter if There was a option to turn off the comment.

I want to move from typeorm to prisma 🥺

**Reply by aqrln** [reply]
I'm not sure if I understand correctly, but do you want to be able to use non-latin characters in the schema and in the generated client code as long as they are valid identifiers in JavaScript/TypeScript? If so, that's not something we currently support (and so just disabling the warning and not creating the comment would not just work), but that would be a valid feature request for us to support it.

**Reply by aqrln** [reply]
Also note that the experience for non-latin fields with `@map` attributes after re-introspection should be relatively smooth after https://github.com/prisma/prisma/issues/19252 is fixed.

**Comment by aqrln** [comment, marked as answer]
By the way I only now noticed that you got a model without name — sorry for missing that the last time. This is definitely a bug. I'll open an issue for that.

**Reply by aqrln** [reply]
https://github.com/prisma/prisma/issues/19307
